src/effect.js

```diff
@@ -0,0 +1,112 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+
+	p5.Effect = function(type) {
+		//everything points to the same audiocontext
+		this.type = type;
+		
+		this.ac = p5sound.audiocontext;
+
+		//wrap each node in gain nodes
+		this.input = this.ac.createGain();
+		this.output = this.ac.createGain();
+
+		this.dry = this.ac.createGain();
+		this.wet = this.ac.createGain();
+
+
+		// this.input.connect(this.output);
+		//connect input to wet and dry gain nodes
+		// this.input.connect(this.dry);
+		// this.input.connect(this.wet);
+
+		// //connect wet and dry gain nodes to output
+		// this.wet.connect(this.output);
+		// this.dry.connect(this.output);
+
+		this.input.connect(this.output);
+
+		//connects to audiocontext destination
+		this.connect();
+
+		if (type) {
+			// determine the type of effect?
+			console.log(type);
+		}
+
+		p5sound.soundArray.push(this);
+	};
+
+
+	p5.Effect.prototype.connect = function (unit) {
+		var u = unit || p5.soundOut.input;
+		this.output.connect(u.input ? u.input : u);
+	};
+
+	p5.Effect.prototype.disconnect = function() {
+		this.output.disconnect();
+	};
+
+	p5.Effect.prototype.dispose = function() {
+		var index = p5sound.soundArray.indexOf(this);
+		p5sound.soundArray.splice(index, 1);
+
+		this.input.disconnect();
+		this.input = undefined;
+
+		this.output.disconnect();
+		this.output = undefined;
+	};
+
+
+	//effects necessary for all effects
+
+
+	// //set up the connection to src and output
+	// p5.Effect.prototype.process = function(src, amt) {
+	// 	src.connect(this.input);
+
+	// 	//set the wetdry value
+	// 	//this.wetdry(amt);
+	// };
+
+/*
+	//sets wet dry value, input double between 0 and 1
+	p5.Effect.prototype.wetdry = function(value) {
+		//eventually implement cross fading like Tone.js
+
+		//for now
+
+		this.dry.value = 1-value;
+		this.wet.value = 1;
+	};
+*/
+	//could link together a bunch of nodes
+	//TODO: find out how to have an open ended number of arguments
+	// p5.Effect.prototype.chain = function(...nodes){
```

Comment:
ooh, cool idea...  `...` is a spread operator that is new in ES6. Until we're able to transpile ES6 to ES5 for compatibility,  you could accomplish this with [`arguments`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)

`chain` seems like it would be better off as a separate PR, once the baseline Effect class is done.